packages/vdom-extension/src/index.ts

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import { JupyterFrontEnd, ILayoutRestorer } from '@jupyterlab/application';
+
+import { InstanceTracker } from '@jupyterlab/apputils';
+
+import { MimeDocumentFactory, MimeDocument } from '@jupyterlab/docregistry';
+
+import { INotebookTracker } from '@jupyterlab/notebook';
+
+import { IRenderMimeRegistry } from '@jupyterlab/rendermime';
+
+import { RenderedVDOM } from './widget';
+
+/**
+ * The CSS class for a VDOM icon.
+ */
+const CSS_ICON_CLASS = 'jp-MaterialIcon jp-VDOMIcon';
+
+/**
+ * The MIME type for VDOM.
+ */
+export const MIME_TYPE = 'application/vdom.v1+json';
+
+/**
+ * The name of the factory that creates VDOM widgets.
+ */
+const FACTORY_NAME = 'VDOM';
+
+const plugin = {
+  id: '@jupyterlab/vdom-extension:factory',
+  requires: [IRenderMimeRegistry, INotebookTracker, ILayoutRestorer],
+  autoStart: true,
+  activate: (
+    app: JupyterFrontEnd,
+    rendermime: IRenderMimeRegistry,
+    notebooks: INotebookTracker,
+    restorer: ILayoutRestorer
+  ) => {
+    const tracker = new InstanceTracker<MimeDocument>({
+      namespace: 'vdom-widget'
+    });
+
+    // Add a renderer factory to application rendermime registry.
+    rendermime.addFactory(
+      {
+        safe: false,
+        mimeTypes: [MIME_TYPE],
+        createRenderer: options => new RenderedVDOM(options)
+      },
+      0
+    );
+
+    notebooks.widgetAdded.connect((sender, panel) => {
```

Comment:
Thanks to @jasongrout for this. This is the secret to providing a notebook's `context` to a rendermime factory.